usr/lib/pkcs11/ep11_stdll/ep11_specific.c

```diff
@@ -7692,3 +7706,194 @@ CK_BBOOL ep11tok_optimize_single_ops(STDLL_TokData_t *tokdata)
 
     return ep11_data->optimize_single_ops ? CK_TRUE : CK_FALSE;
 }
+
+/* return -1 if v1 < v2, 0 if v1 == v2, 1 if v1 > v2 */
+static int compare_ck_version(CK_VERSION *v1, CK_VERSION *v2)
+{
+
+    if (v1->major < v2->major)
+        return -1;
+    if (v1->major > v2->major)
+        return 1;
+    if (v1->minor < v2->minor)
+        return -1;
+    if (v1->minor > v2->minor)
+        return 1;
+    return 0;
+}
+
+static CK_RV get_card_type(uint_32 adapter, CK_ULONG *type)
+{
+    char fname[290];
```

Comment:
290 -> PATH_MAX

Reply:
Will be fixed. I will update the patch when you are through with your review.